src/priority.c

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2022 Joel E. Anderson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <stumpless.h>
+#include "private/memory.h"
+#include "private/entry.h"
+
+static void toUpperCase( char *str ) {
+  for( int i = 0; i < str[i]; i++) {
+    str[i] = toupper( str[i] );
+  }
+}
+
+int
+stumpless_prival_from_string( const char *string ) {
+  int prival;
+  int severity;
+  int facility;
+  char *param;
+  char *period;
+  char *sec_period;
+  size_t len;
+  size_t slen;
+  size_t pre_len;
+  const char pre_facility[] = "STUMPLESS_FACILITY_";
+  const char pre_severity[] = "STUMPLESS_SEVERITY_";
+
+  if( !string || !string[0] )
```

Comment:
use `VALIDATE_ARG_NOT_NULL_INT_RETURN` to make sure that the argument is not NULL here, so that the error code is correctly set

Reply:
done